proposals/nnnn-mutex.md

```diff
@@ -310,9 +303,13 @@ func something() {
 
 By marking the closure as such, we've effectively declared that the mutex is in itself its own isolation domain. We must not let non-sendable values it holds onto be unsafely sent across isolation domains to prevent these holes of shared mutable state.
 
-### When do I use `Mutex` vs. an Actor?
+### Differences between mutexes and actors
+
+The mutex type we're proposing is a synchronous lock. This means when other participants want to acquire the lock to access the protected shared data, they will hault execution until they are able to do so. Those that are waiting to acquire the lock will not be able to make forward progress until their request to acquire the lock has completed. This can lead to thread contention if the acquired thread's critical section is not able to be executed relatively quickly exhausting resources for the rest of the system to continue making forward progress. Traditional synchronous locks are also prone to deadlocks and livelocks which can leave a process in an unrecoverable state. These scenarios can occur when there is a complex hierarchery of different locks that manage to depend on the acquisition of each other.
```

Comment:
```suggestion
The mutex type we're proposing is a synchronous lock. This means when other participants want to acquire the lock to access the protected shared data, they will halt execution until they are able to do so. Threads that are waiting to acquire the lock will not be able to make forward progress until their request to acquire the lock has completed. This can lead to thread contention if the acquired thread's critical section is not able to be executed relatively quickly exhausting resources for the rest of the system to continue making forward progress. Synchronous locks are also prone to deadlocks (which Swift's actors cannot currently encounter due to their re-entrant nature) and live-locks which can leave a process in an unrecoverable state. These scenarios can occur when there is a complex hierarchy of different locks that manage to depend on the acquisition of each other.
```